src/components/organisms/ProfileModal/UserInformationContent/UserInformationContent.tsx

```diff
@@ -1,179 +1,214 @@
-    () =>
-      profile
-        ? profileQuestions?.map((question) => {
-            // @ts-ignore User type doesn't accept string indexing. We need to rework the way we store answers to profile questions
-            const questionAnswer = profile[question.name];
-
-            if (!questionAnswer) return undefined;
-
-            return (
-              <div key={question.name} className="question-section">
-                <div className="question">{question.text}</div>
-                <div className="answer">{questionAnswer}</div>
-              </div>
-            );
-          })
-        : undefined,
-    [profile, profileQuestions]
+  }, [user]);
+
+  const editProfileLink = useCallback(
+    (profileLink) => () => {
+      setProfileLinkToEdit(profileLink);
+      setUserProfileMode(UserProfileMode.EDIT_PROFILE_LINK);
+    },
+    [setProfileLinkToEdit, setUserProfileMode]
   );
 
-  if (!user || !userWithId) return null;
+  if (!user) return null;
 
   return (
-    <>
-      <h1 className="title modal-title">My Profile</h1>
-
-      <div className="user-information">
-        <UserAvatar user={userWithId} showStatus large />
-        <div className="text-container">
-          <h3>{profile?.partyName || DEFAULT_PARTY_NAME}</h3>
-          <div className="ellipsis-text">{user.email}</div>
-          <UserStatusDropdown />
+    <div className="UserInformationContent">
+      <h1 className="UserInformationContent__title">My Profile</h1>
+      <div className="UserInformationContent__information">
+        <div>
+          <UserAvatar user={user} showStatus large />
         </div>
+        <div className="UserInformationContent__text-container">
+          <h3 className="UserInformationContent__user-name">
+            {user?.partyName ?? DEFAULT_PARTY_NAME}
+          </h3>
+          <div
+            title={email ?? ""}
+            className="UserInformationContent__ellipsis-text"
+          >
+            {email}
+          </div>
+          <div className="UserInformationContent__status-container">
+            <span className="UserInformationContent__status-prefix">
+              Available
+            </span>
+            <UserStatusDropdown />
+          </div>
+        </div>
+        <Button
+          customClass="UserInformationContent__edit"
+          onClick={() => setUserProfileMode(UserProfileMode.EDIT_PROFILE)}
+        >
+          <img src={editIcon} alt="edit" />
+        </Button>
       </div>
-
-      {renderedProfileQuestionAnswers}
+      {profileQuestions &&
+        profileQuestions.map((question: QuestionType) => (
+          <div key={question.name} className="question-section">
+            <div className="question">{question.text}</div>
+            <div>
+              {
+                // @ts-ignore question.name is a correct index for type User
+                (profile && profile[question.name]) ||
+                  DEFAULT_PROFILE_QUESTION_ANSWER
+              }
```

Comment:
```suggestion
                (profile?[question.name]) ||
                  DEFAULT_PROFILE_QUESTION_ANSWER
              }
```

Reply:
FYI: I think it needs to be `profile?.[question.name]`

Reply:
fixed: https://github.com/sparkletown/sparkle/pull/1499/commits/4b9c4fee3c3c8ac3d75c4a18814995447bfa29da#diff-9f9d7328f3ef637d17d442f038b2da17f90a06672d7fb0845d12e066305ee141R148-R156